### PR TITLE
Add support for PYTHON_VERSION env variable.

### PR DESCRIPTION
**Pull request by duckinator**
My attempt at implementing #362.

I wasn't able to test it, unfortunately, because steps 2 and/or 3 in the README seem to be incorrect. There's no `build` command available in the Docker image.

### CODE REVIEW

**Comment by duckinator** [comment]
Resolved the conflicts caused by Swift support being added.

**Comment by duckinator** [comment]
I also have no idea what's up with this:

Screenshot_20200429_210138

**Comment by vbrown608** [comment]
@duckinator it means that Netlify's backend engineering team was automatically notified that we need to review your PR. I'll take a look and follow up soon. Thanks for your contribution!

File: run-build-functions.sh [hunk 1]
```diff
@@ -184,7 +185,7 @@ install_dependencies() {
     echo "Found Pipfile restoring Pipenv virtualenv"
     restore_cwd_cache ".venv" "python virtualenv"
   else
-    source $HOME/python2.7/bin/activate
```
Comment (vbrown608) [inline run-build-functions.sh:192]: The Docker container this runs in only has python 2.7, 3.5, and 3.7 installed. If `PYTHON_VERSION` isn't one of those, we should do some error handling like we're doing  when the user tries to load the version from `runtime.txt`.

**Review by vbrown608 — CHANGES_REQUESTED** [review]
This is a nice improvement, thanks! One small comment about error handling but otherwise looking good.

For local testing - I'm able to run the `build` command in the Docker image as described by the README, though it errors with the following because of https://github.com/netlify/build-image/issues/402 
```
Version '--no-progress' not found - try `nvm ls-remote` to browse available versions.
```
Opening a PR to address that, in the meantime you can edit the last line of `test-tools/start-image.sh` as the submitter suggests.

If you're still having trouble, could you try making sure you've pulled the most recent version of the `netlify/build:xenial` image? I'm also happy to test the PR locally before merging :)

**Comment by duckinator** [comment]
Rebased off `xenial`; gonna try to address @vbrown608's review now.

**Comment by duckinator** [comment]
@vbrown608 I think this is ready for another look! :slightly_smiling_face:

**Comment by erans** [comment]
+1 on being able to specify PYTHON_VERSION as an env var and (even more important to me) - being able to support Python 3.8!

**Review by vbrown608 — APPROVED** [review]
Sorry for the delay, tested this in our build system and it's looking good. Thanks for your contribution!

**Comment by erans** [comment]
> Sorry for the delay, tested this in our build system and it's looking good. Thanks for your contribution!

Does it mean you can just control it via the env var or does it also mean there is Python 3.8 support?

**Comment by vbrown608** [comment]
This PR adds the ability to control the python version through an env var but it must still be a supported version. It doesn't add support for 3.8.

**Comment by revolter** [comment]
This doesn't work :( I set `PYTHON_VERSION` to `3.7`, but it still uses `2.7`.

**Comment by revolter** [comment]
Using

```toml
[build.environment]
  PYTHON_VERSION = "3.7"
```

in a `netlify.toml` file doesn't work either :(

https://app.netlify.com/sites/fastlane-docs-preview/deploys/61991e5e6689420008c5416c

**Comment by duckinator** [comment]
@revolter the documentation says only 2.7 and 3.8 are supported: https://github.com/netlify/build-image/blob/focal/included_software.md#languages

It seems for Python this isn't installing the language runtime on-the-fly, it's letting you choose from two installed versions. And in this case, the docs imply 3.8 will be used by default. So just removing that entirely might be your best bet.

**Comment by duckinator** [comment]
Although the fact that it claims to default to 3.8, but falls back to 2.7 if it can't find the specified version is... bizarre...

**Comment by revolter** [comment]
> @revolter the documentation says only 2.7 and 3.8 are supported: focal/included_software.md#languages

This was merged in the branch `xenial`, which means that the correct documentation is this: https://github.com/netlify/build-image/blob/xenial/included_software.md#languages, so `3.7` is supported.

> Although the fact that it claims to default to 3.8, but falls back to 2.7 if it can't find the specified version is... bizarre...

Same thing as above. On Xenial, `2.7` is the default.

**Pull request merged**